components/Paginator/index.jsx

```diff
@@ -82,7 +78,8 @@ export default class Paginator extends React.Component {
       query
     } = this.props;
 
-    const queryObject = queryParamsToObject(query);
+    const queryObject = (typeof query === 'object') ? query : queryString.parse(query);
```

Comment:
maybe create a classMethod to generate the queryObject, then you don't need to write the same code twice